src/libcollections/vec.rs

```diff
@@ -1505,6 +1505,22 @@ impl<'a> From<&'a str> for Vec<u8> {
 // Clone-on-write
 ////////////////////////////////////////////////////////////////////////////////
 
+#[stable(feature = "rust1", since = "1.7.0")]
+impl<'a, T: Clone> From<&'a [T]> for Cow<'a, [T]> {
+    #[inline]
```

Comment:
Can you remove these for now? The generic aspect of this already provides cross-crate inlining and after that we don't need much coercion to get LLVM to inline it.